### DISCUSSION TITLE
Add header options field for OpenAI compatible API provider

### DISCUSSION THREAD

**Discussion by atmaker** [cline/cline#734] [Feature Requests]
### Problem
I'd like to use custom models from my Open WebUI instance with Cline, but this is currently not possible, because OpenWebUI's API requires a custom authentication header to be sent with the request.

### Solution
I'm proposing a separate settings field for OpenAI compatible API provider in Cline which would allow you to set the header options. This would solve the authentication problem, and it could be useful also in other regards.

**Comment by just-maik** [comment]
Yes, I need this too!

**Comment by chandan84** [comment]
my company provides access to models on azure, but behind an api management gateway, that requires a custom authentication token in header and some static ones for authorization as well. adding some support around this would be super helpful as well
